src/mesh_routines.f90

```diff
@@ -1682,12 +1682,12 @@ SUBROUTINE DECOMP_TOPOLOGY_ELEM_ADJACENT_ELEM_CALCULATE(TOPOLOGY,ERR,ERROR,*)
     INTEGER(INTG) :: j,ne,ne1,nep1,ni,nic,nn,nn1,nn2,nn3,np,np1,DUMMY_ERR,FACE_XI(2),FACE_XIC(3),NODE_POSITION_INDEX(4)
     INTEGER(INTG) :: xi_direction,direction_index,xi_dir_check,xi_dir_search,NUMBER_NODE_MATCHES
     INTEGER(INTG) :: candidate_idx,face_node_idx,node_idx,surrounding_el_idx,candidate_el,idx
-    INTEGER(INTG) :: SURROUNDING_ELEMENTS(100) !Fixed size array... should be large enough
-    INTEGER(INTG) :: NUMBER_SURROUNDING,NUMBER_OF_NODES_XIC(4)
+    INTEGER(INTG) :: NUMBER_SURROUNDING,NUMBER_OF_NODES_XIC(4), count, i
     INTEGER(INTG), ALLOCATABLE :: NODE_MATCHES(:),ADJACENT_ELEMENTS(:)
     LOGICAL :: XI_COLLAPSED,FACE_COLLAPSED(-3:3),SUBSET
     TYPE(LIST_TYPE), POINTER :: NODE_MATCH_LIST
     TYPE(LIST_PTR_TYPE) :: ADJACENT_ELEMENTS_LIST(-4:4)
+    TYPE(LIST_TYPE), POINTER :: SURROUNDING_ELEMENTS
```

Comment:
You mean like this?

INTEGER(INTG), ALLOCATABLE :: NODE_MATCHES(:),ADJACENT_ELEMENTS(:)
LOGICAL :: XI_COLLAPSED,FACE_COLLAPSED(-3:3),SUBSET
TYPE(LIST_TYPE), POINTER :: SURROUNDING_ELEMENTS
TYPE(LIST_TYPE), POINTER :: NODE_MATCH_LIST

On Fri, 24 Apr 2015 at 14:44 Chris Bradley <notifications@github.com> wrote:

> In src/mesh_routines.f90
> <https://github.com/OpenCMISS/cm/pull/198#discussion_r29020556>:
>
> >      INTEGER(INTG), ALLOCATABLE :: NODE_MATCHES(:),ADJACENT_ELEMENTS(:)
> >      LOGICAL :: XI_COLLAPSED,FACE_COLLAPSED(-3:3),SUBSET
> >      TYPE(LIST_TYPE), POINTER :: NODE_MATCH_LIST
> >      TYPE(LIST_PTR_TYPE) :: ADJACENT_ELEMENTS_LIST(-4:4)
> > +    TYPE(LIST_TYPE), POINTER :: SURROUNDING_ELEMENTS
>
> It might be tidier to add SURROUNDING_ELEMENTS to the type declaration two
> lines above.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenCMISS/cm/pull/198/files#r29020556>.
>